### PR TITLE
cortextool: init at version 0.1.4

### PR DESCRIPTION
**Pull request by ekpdt**
###### Motivation for this change

Add `cortextool`, a CLI for interacting with [Cortex](http://cortexmetrics.io/) instances.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kalbasit** [comment]
aarch64 is failing. Please restrict this package to linux/darwin. doc :point_right: https://nixos.org/nixpkgs/manual/#chap-meta

**Comment by ekpdt** [comment]
Is it possible this is a generalized problem with the go packaging rather than a platform specific issue? Both `aarch64-linux` and `x86_64-linux` fail with a complaint about the `+incompatible` suffix [introduced in go 1.13](https://golang.org/doc/go1.13#modules)
`k8s.io/client-go@v12.0.0+incompatible: invalid version: +incompatible suffix not allowed: module contains a go.mod file, so semantic import versioning is required`

**Comment by ekpdt** [comment]
I was able to build this package by apply this PR to nixpkgs back when it used go 1.12 (not 1.13 or 1.14). So it looks like the `go.mod` in the upstream sources is incompatible with more recent versions of go.

**Comment by ekpdt** [comment]
The upstream project uses `-mod=vendor` in conjunction with `go build`, not `go mod`. See https://github.com/grafana/cortextool/blob/master/Makefile
It's a bit beyond me at the moment to figure out how (or even if) to shim this into the `buildGoModule` machinery.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2248,6 +2248,12 @@
     githubId = 411447;
     name = "Leo Gaspard";
   };
+  ekpdt = {
+    email = "nix@pdtpartners.com";
+    github = "ekpdt";
+    githubId = 64796365;
+    name = "Elan Kugelmass";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2255]: Please put this into a separate commit.

File: pkgs/servers/monitoring/cortex/cortextool.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+buildGoModule rec {
+  pname = "cortextool";
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "grafana";
+    repo = "cortextool";
+    sha256 = "0p3g418bg6agk6manibazlb2rplv8253z5gka5nb37jvccz6k38v";
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/cortex/cortextool.nix:11]:
```suggestion
    owner = "grafana";
    repo = "cortextool";
    rev = "v${version}";
    sha256 = "0p3g418bg6agk6manibazlb2rplv8253z5gka5nb37jvccz6k38v";
```

File: pkgs/servers/monitoring/cortex/cortextool.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "A powerful command line tool for interacting with cortex";
+    homepage = "https://github.com/grafana/cortextool";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ekpdt ];
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/cortex/cortextool.nix:34]:
```suggestion
}

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
```
nixpkgs/pkgs/development/go-modules/generic/default.nix:3:1 called without required argument 'vendorSha256'
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ekpdt) [reply maintainers/maintainer-list.nix:2255]: Same PR, separate commit
or
Different PR
?

**Comment by SuperSandro2000** [comment]
> ```
> nixpkgs/pkgs/development/go-modules/generic/default.nix:3:1 called without required argument 'vendorSha256'
> ```

Please fix the eval error.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:2255]: > Same PR, separate commit

File: pkgs/servers/monitoring/cortex/cortextool.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "grafana";
+    repo = "cortextool";
+    rev = "v${version}";
+    sha256 = "0p3g418bg6agk6manibazlb2rplv8253z5gka5nb37jvccz6k38v";
+  };
+
+  modSha256 = "0wpmn6clmsb5ird9h8wx6njg6xgpmw80zx4dbpmifsgirldi5yjd";
+
+  buildFlagsArray =''
+    -ldflags=
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/cortex/cortextool.nix:17]:
```suggestion
    -ldflags=
        -s -w
```

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.
